config/agents.go

```diff
@@ -47,7 +48,16 @@ func (ac AgentConfig) ContainerName() string {
 	if ac.IsLocal {
 		return fmt.Sprintf("%s-agent-%s", ContainerNamePrefix, utils.ShortenString(ac.ID, 8))
 	}
-	return fmt.Sprintf("%s-agent-%s-%s", ContainerNamePrefix, utils.ShortenString(ac.ID, 8), utils.ShortenString(digest, 4))
+	return fmt.Sprintf(
+		"%s-agent-%s-%s", ContainerNamePrefix, utils.ShortenString(ac.ID, 8), utils.ShortenString(digest, 4),
+	)
+}
+
+func (ac AgentConfig) Compare(b AgentConfig) bool {
```

Comment:
Nice! I really like methods that encapsulate this kind of complicated checks. Can we rename `Compare` to `IsEqual` or something? For example, I find `if !cfg.IsEqual()` easier to understand than `if !cfg.Compare()`.